object-mapper.rst

```diff
@@ -0,0 +1,260 @@
+Object Mapping
+==============
+
+Symfony provides a mapper to transform a given object to another one.
+
+.. _activating_the_serializer:
+
+Installation
+------------
+
+Run this command to install the ``object-mapper`` before using it:
+
+.. code-block:: terminal
+
+    $ composer require symfony/object-mapper
+
+Using the ObjectMapper Service
+------------------------------
+
+Once enabled, the object mapper service can be injected in any service where
+you need it or it can be used in a controller::
+
+    // src/Controller/DefaultController.php
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\ObjectMapper\ObjectMapperInterface;
+
+    class DefaultController extends AbstractController
+    {
+        public function index(ObjectMapperInterface $objectMapper): Response
+        {
+            // keep reading for usage examples
+        }
+    }
+
+
+Map an object to another one
+----------------------------
+
+To map an object to another one use ``map``::
+
+    use App\Dto\Source;
+    use App\Dto\Target;
+
+    $mapper = new ObjectMapper();
+    $mapper->map(source: new Source(), target: Target::class);
+
+
+If you alread have a target object, you can use its instance directly::
+
+    use App\Dto\Source;
+    use App\Dto\Target;
+
+    $target = new Target();
+    $mapper = new ObjectMapper();
+    $mapper->map(source: new Source(), target: $target);
+
+
+Configure the mapping target using attributes
+---------------------------------------------
+
+The Object Mapper component includes a :class:`Symfony\\Component\\ObjectMapper\\Attribute\\Map` attribute to configure mapping
+behavior between objects. Use this attribute on a class to specify the
+target class::
+
+    // src/Dto/Source.php
+    namespace App\Dto;
+
+    use Symfony\Component\ObjectMapper\Attributes\Map;
+
+    #[Map(target: Target::class)]
+    class Source {}
+
+Configure property mapping
+--------------------------
+
+Use the :class:`Symfony\\Component\\ObjectMapper\\Attribute\\Map` attribute on properties to configure property mapping between
+objects. ``target`` changes the target property, ``if`` allows to
+conditionally map properties::
+
+    // src/Dto/Source.php
+    namespace App\Dto;
+
+    use Symfony\Component\ObjectMapper\Attributes\Map;
+
+    class Source {
+        #[Map(target: 'fullName')]
+        public string $firstName;
+
+        #[Map(if: false)]
+        public string $lastName;
+    }
+
+Transform mapped values
+-----------------------
+
+Use ``transform`` to call a function or a
+:class:`Symfony\\Component\\ObjectMapper\\CallableInterface`::
+
+    // src/ObjectMapper/TransformNameCallable.php
+    namespace App\ObjectMapper;
+
+    use App\Dto\Source;
+    use Symfony\Component\ObjectMapper\CallableInterface;
+
+    /**
+     * @implements CallableInterface<Source>
+     */
```

Comment:
Is this necessary in a documentation example ?

Reply:
I think that its best to have a copy/paste that doesn't trigger phpstan errors